### PR TITLE
Feature: I want to automatically switch to the last keyboard language selected in the conversation

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Add language attribute to the conversation entity and create tests.

### CODE REVIEW

File: Source/Model/Conversation/ZMConversation+Language.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public extension ZMConversation {
+    @NSManaged internal var language: String
+
+    public var inputLanguage: String? {
```
Comment (mikeger) [inline Source/Model/Conversation/ZMConversation+Language.swift]: Can you explain the necessity of having two properties? Can't we just expose the CoreData property?

File: Resources/zmessaging.xcdatamodeld/zmessaging2.50.0.xcdatamodel/contents [hunk 2]
```diff
@@ -47,6 +47,7 @@
         <attribute name="internalIsArchived" attributeType="Boolean" defaultValueString="NO" usesScalarValueType="NO" syncable="YES"/>
         <attribute name="isSelfAnActiveMember" attributeType="Boolean" defaultValueString="YES" usesScalarValueType="NO" syncable="YES"/>
         <attribute name="isSilenced" attributeType="Boolean" defaultValueString="NO" usesScalarValueType="NO" syncable="YES"/>
+        <attribute name="language" optional="YES" attributeType="String" syncable="YES"/>
```
Comment (mikeger) [inline Resources/zmessaging.xcdatamodeld/zmessaging2.50.0.xcdatamodel/contents]: Have you tested that the app still migrating correctly?

Comment (vytis) [reply Resources/zmessaging.xcdatamodeld/zmessaging2.50.0.xcdatamodel/contents]: But 2.50 is already merged to develop and is used in internal clients. You have to create a new model version and add a migration test to make sure it all works fine.

File: Tests/Source/Model/Conversation/ZMConversationTests+Language.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+class ZMConversationTests_Language : BaseZMMessageTests {
+
+    func testThatItAllowsSettingLanguageOnConversation(){
+        // given
+        let conversation = ZMConversation.insertNewObject(in: uiMOC)
+        let germanLanguage = "de-DE"
+
+        // when
+        conversation.inputLanguage = germanLanguage
```
Comment (vytis) [inline Tests/Source/Model/Conversation/ZMConversationTests+Language.swift]: A more thorough test would be to insert the object, change the language and the re-fetch it again. The current test would pass even if inputLanguage would be transitive property and never saved to the store.

**Comment by billypchan** [comment]
@zenkins release minor

**Comment by billypchan** [comment]
@zenkins release minor

**Pull request merged**